src/main/java/com/github/wovnio/wovnjava/Headers.java

```diff
@@ -27,7 +27,8 @@
         this.request = r;
         this.urlLanguagePatternHandler = urlLanguagePatternHandler;
 
-        this.requestLang = this.computeRequestLang();
+        String clientRequestUrl = UrlResolver.computeClientRequestUrl(r, s);
```

Comment:
definitely rename r and s parameters

Reply:
Not a huge fan of making it static though, it's the kind of thing I think should be passed in as a parameter to Headers, `new UrlResolver(request, settings)`

Reply:
I will ask to keep UrlResolver as a static class for now. The reason is that it keeps this set of changes short and concise.

In essence, we currently both compute request URL and find its request language inside `Headers.java`. Now, the "compute request URL" part is moved outside and tested separately, but otherwise `Headers.java` has the same purpose and design as before.

`Headers.java` will still receive a lot of changes, and how to best combine the components will become easier to see as we clean up the old code.